docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,475 @@
+      <!-- For accessing external storage -->
+      <privilege>http://tizen.org/privilege/externalstorage</privilege>
+    </privileges>
+    ```
+
+## Building blocks
+
+Following are the four major components of Machine Learning Training API:
+
+ - [Model](#model)
+ - [Layer](#layer)
+ - [Optimizer](#optimizer)
+ - [Dataset](#dataset)
+
+### Model
+
+
+Model is a wrapper component that has the topology of layers, optimizers, and datasets.
+The model performs training and saves the updated parameters that can later be used for inference.
+
+![model](./media/model.png)
+
+In this figure, `data` represent input data (or feature) and `label` is the actual value to be tested over prediction.
+Currently, only a sequential neural network is supported:
+
+```c
+// Create model
+ml_train_model_h model;
+ml_train_model_construct(&model);
+
+/* Configure model(omitted for brevity) */
+
+// Compile model. This freezes model and afterwards the model cannot be modified.
+ml_train_model_compile(model, "loss=cross", "batch_size=16", NULL);
+// run model
+ml_train_model_run(model, "epochs=2", "save_path=model.bin", NULL);
+
+// destroy after use
+ml_train_model_destroy(model);
+```
+
+A number of properties can be set at `ml_train_model_compile()` and `ml_train_model_run()` phase:
+
+Function | Key | Value | Description
+-------- | --- | ----- | -----------
+`ml_train_model_compile()` | loss      | (categorical)| Loss function to be set
+&#xfeff;                 |           | cross        | cross-entropy loss
+&#xfeff;                 |           | mse          | mean squared error
+`ml_train_model_run()`     | epochs    | (integer)    | Determines epochs for the model
+&#xfeff;                 | save_path | (file_path)  | Model path to save and read parameters
+
+### Layer
+
+Layer is a component that does actual computation while managing internal trainable parameters.
+Currently, input layer and fully connected layer type are supported:
+
+```c
+// Create layer
+ml_train_layer_h layer;
+ml_train_layer_create(&layer, ML_TRAIN_LAYER_TYPE_FC);
+
+// configure layer
+ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=zeros", NULL);
+
+// after adding the layer to model,
+// you do not need to destroy layer since ownership is transferred to the model.
+ml_train_model_add_layer(model, layer);
+```
+
+Following are the available properties for each layer type:
+
+Type | Key | value | Description
+---------- | --- | ----- | -----------
+(Universal properties)           |                    |               | Universal properties that applies to every layer
+&#xfeff;                    | name               | (string)      | An identifier for each layer
+&#xfeff;                    | input_shape        | (string)      | Formatted string as "channel:height:width". (if there is no channel it should be 1). First layer of the model must have input_shape. Other can be omitted as it is calculated at compile phase.
```

Comment:
Suggestion:

Formatted string as "channel:height:width" (if there is no channel then it must be 1).

Please verify. Not clear.

Reply:
Done. I also removed the parenthesis. 

`Formatted string as "channel:height:width". If there is no channel then it must be 1.`